### ISSUE TITLE
Is writing a WINE dockerfile a version worth it right now?

### ISSUE THREAD

**Issue by pushfoo** [Creatures-Developer-Network/docker-ds#5]
@Kirtai had tried experimenting with WINE earlier. If we can get it working, it might be a cheaper alternative than constantly testing against cloud instance options. A [WINE base image exists](https://github.com/scottyhardy/docker-wine) exists, and under a permissive MIT license. DS on my Windows 10 virtualbox VM seems to constantly break the video for unclear reasons, and others might experience the same thing.

At the same time, from what Kirtai said, it seems like the shared memory interface might not work under WINE. There are also long-run issues such whether to have a base DS images that has an environment manager, but that could be saved for later. Mounting volumes could work perfectly fine for now.

If it's best to investigate the shared memory interface and its issues before putting in the effort to dockerize DS under WINE, we could do that first.

**Comment by robinduckett** [comment]
I guess we should come up with concrete testing examples in order to test the client and some debug app which uses the shared memory interface to retrieve some data - perhaps some kind of unit testing suite as a precursor to fuzzing?

**Comment by pushfoo** [comment]
> perhaps some kind of unit testing suite as a precursor to fuzzing?

Yes. That's why i was so eager to get pyc2e written. I saw it as the first stepping stone to a framework we can use to manage instances and fuzzing.
I have a ticket on prayer that i need to finish but i'll get back to pyc2e and openc2e interface work soon. As i mentioned in discord, I think a model for improvements could be:
1. prototype a feature where it's first needed, in either JS or python
2. implement it in c/c++ in openc2e
3. add bindings for the c/c++ everywhere else, so all end-user tools have access to it

**Comment by pushfoo** [comment]
[This post](https://web.archive.org/web/20161119032049/https://sheeteashop.tumblr.com/osx-lion) might be useful for getting c2e working in WINE, even though it's years out of date. I discovered it while trying to track down who might have backups of old Lifandi concept art, and I figure it's best to post it somewhere public in case someone else needs it.
